chromiumcontent/BUILD.gn

```diff
@@ -0,0 +1,376 @@
+      complete_static_lib = true
+      deps = [ "//third_party/libyuv" ]
+  }
+
+  static_library("ppapi_cpp_objects") {
+      complete_static_lib = true
+      deps = [ "//ppapi/cpp:objects" ]
+  }
+
+  static_library("ppapi_internal_module") {
+      complete_static_lib = true
+      deps = [ "//ppapi/cpp/private:internal_module" ]
+  }
+
+  static_library("security_state") {
+      complete_static_lib = true
+      deps = [ "//components/security_state" ]
+  }
+
+  static_library("system_wrappers") {
+      complete_static_lib = true
+      deps = [ "//third_party/webrtc/system_wrappers" ]
+  }
+
+  static_library("webrtc_common") {
+      complete_static_lib = true
+      deps = [ "//third_party/webrtc:webrtc_common" ]
+  }
+
+  if (is_linux) {
+    # The original libgtk2ui target is a shared library, so we must list
+    # the object files instead of just depending on it.
+    static_library("libgtk2ui") {
+      complete_static_lib = true
+      sources = [
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/app_indicator_icon.o",
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/app_indicator_icon_menu.o",
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/gtk2_util.o",
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/gtk2_status_icon.o",
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/menu_util.o",
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/skia_utils_gtk2.o",
+        "$root_out_dir/obj/chrome/browser/ui/libgtk2ui/libgtk2ui/unity_service.o",
+      ]
+    }
+  }
+
+  if (is_win) {
+    static_library("sandbox_helper_win") {
+      complete_static_lib = true
+      deps = [ "//content:sandbox_helper_win" ]
+    }
+  }
+}
+
+if (is_electron_build && !is_component_build) {
+
+  # This is where we build statically linked libchromiumcontent.
+  # After all dependent targets are built, we use a script to collect
+  # all required object files into an "objects.gni" file and re-run
+  # ninja. Because "objects.gni" is a GN input file, it will cause it
+  # to re-generate the targets and produce the static libraries we want.
+
+  # Make sure the .gni file exists, otherwise it cannot be imported
+  if (getenv("CHROMIUMCONTENT_2ND_PASS") == "") {
+    write_file("$target_out_dir/objects.gni", "")
+  }
+
+  import("$target_out_dir/objects.gni")
+
+  action("chromiumcontent") {
+    outputs = [ "$target_out_dir/build_libs.done" ]
+
+    script = "build_libs.py"
+    args = [ "-o", rebase_path("$target_out_dir/objects.gni"),
+             "-s", rebase_path(outputs[0]) ]
+
+    deps = [ ":targets" ]
+  }
+
+  # Normally we would just put everything in a single library plus V8. But
+  # toolchains have various limitations that we would hit - e.g. MSVC cannot
+  # produce a .lib file bigger than 4 GB, LLVM has a limit on the number of
+  # object files on the command line. So we produce multiple libraries.
+  group("libs") {
```

Comment:
Interesting. 👍 If your patch gets merged to Chromium, we can surely combine some of the libs. Right now I don't see a benefit which would warrant adding this patch to libcc. And the MSVC limitation remains.